### PR TITLE
MultiServer: send new read_hints datastore values on change

### PR DESCRIPTION
**Pull request by Berserker66**
## What is this fixing or adding?
send new read_hints datastore values on change

## How was this tested?
image

## If this makes graphical changes, please attach screenshots.
image

### CODE REVIEW

**Comment by Jarno458** [comment]
I think for reference this affects #2521

**Review by Exempt-Medic — APPROVED** [review]
The code looks fine to me, tested a few games and the hints worked, including updating on location clears

File: MultiServer.py [hunk 1]
```diff
@@ -973,7 +976,10 @@ def register_location_checks(ctx: Context, team: int, slot: int, locations: typi
             "hint_points": get_slot_points(ctx, team, slot),
             "checked_locations": new_locations,  # send back new checks only
         }])
-
+        old_hints = ctx.hints[team, slot].copy()
+        ctx.recheck_hints(team, slot)
+        if old_hints != ctx.hints[team, slot]:
```
Comment (black-sliver) [inline MultiServer.py:981]: Wouldn't it make sense to use our helper here?
```suggestion
        if old_hints != ctx.get_rechecked_hints(team, slot):
```

Comment (Berserker66) [reply MultiServer.py:981]: Style wise, I don't like "hiding" mutations in an if statement.

Comment (black-sliver) [reply MultiServer.py:983]: should we get rid of the helper then? any use of the helper will hide mutation

Comment (ThePhar) [reply MultiServer.py:983]: Or potentially rename the helper to be more clear what it does.

Comment (Berserker66) [reply MultiServer.py:983]: its only other use currently is outside an if statement.

**Review by black-sliver — APPROVED** [review]
Approving this. We should probably still get rid of the mutating call, or rename it at some point to make it more clear. I don't *really* see mutation in an if as a problem, but in combination with the naming, it's not great for sure.

**Pull request merged**